packages/block-library/src/image/index.js

```diff
@@ -56,4 +57,21 @@ export const settings = {
 	deprecated,
 };
 
-export const init = () => initBlock( { name, metadata, settings } );
+export const init = () => {
+	addFilter(
+		'editor.hooks.updateAlignment',
+		'core/image/update-alignment',
+		( nextAlign, blockName ) =>
+			blockName === 'core/image' &&
+			[ 'wide', 'full' ].includes( nextAlign )
+				? {
+						width: undefined,
+						height: undefined,
+						aspectRatio: undefined,
+						scale: undefined,
+						align: nextAlign,
+				  }
+				: { align: nextAlign }
```

Comment:
I'm not sure about the way the attributes object is constructed in this filter function. With this filter we're passing in one kind of data and receiving a different type, while I always thought of a filter as acting upon a single type of data (e.g. attributes in / attributes out).

My thinking is it could be better to name the filter `updateAlignmentAttributes` and then pass the `{align: nextAlign }` object in. That would also allow the filter function to do an early return without modifying the data:
```js
( blockName, updateAttributes ) => {
    if ( blockName !== 'core/image' ) {
        return updateAttributes;
    }
    // ...
}
```

Also, I think this filter function could be unregistered/removed by a plugin, so that could be dangerous.